.github/dependabot.yml

```diff
@@ -5,19 +5,96 @@ registries:
     url: https://plugins.gradle.org/m2
     username: dummy # Required by dependabot
     password: dummy # Required by dependabot
+  maven-central:
+    type: maven-repository
+    url: https://repo1.maven.org/maven2/
+    username: dummy # Required by dependabot
+    password: dummy # Required by dependabot
 updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    target-branch: "1.9.x" # oldest supported branch
+    schedule:
+      interval: "weekly"
+# Build dependencies; only target the oldest supported branch
   - package-ecosystem: "gradle"
     directory: "/"
-    target-branch: "1.9.x" # oldest OSS supported branch
+    schedule:
+      interval: daily
+    target-branch: "1.9.x"
+    groups:
+      netflix-plugins:
+        patterns:
+          - "com.netflix.nebula*"
+        update-types:
+          - minor
+          - patch
     allow:
+      - dependency-name: "gradle.plugin.com.hierynomus.gradle.plugins*"
+      - dependency-name: "com.netflix.nebula*"
+      - dependency-name: "io.spring.nohttp*"
+      - dependency-name: "io.github.gradle-nexus*"
+      - dependency-name: "io.spring.javaformat*"
+      - dependency-name: "me.champeau.gradle*"
+      - dependency-name: "de.undercouch:gradle*"
+      - dependency-name: "com.diffplug.spotless*"
+      - dependency-name: "biz.aQute.bnd:biz.aQute.bnd.gradle*"
+      - dependency-name: "io.spring.ge.conventions*"
       - dependency-name: "com.gradle*"
+      - dependency-name: "org.gradle*"
+    ignore:
+      # only upgrade minor and patch versions for build dependencies
+      - dependency-name: "*"
+        update-types:
+          - version-update:semver-major
     registries:
       - gradle-plugin-portal
+      - maven-central
+# Non-build dependencies; target every supported branch
+  - package-ecosystem: gradle
+    directory: /gradle
```

Comment:
Dependabot config does not let you define multiple configurations for the same combination of package ecosystem, directory, and target branch. Hence we hack around this by specifying a different directory for "non-build dependencies", which we happen to know will all be defined in the TOML file in the `/gradle` folder. This differentiates it from the configuration for the "build dependencies" which uses `/` as the directory because they will be defined in both `settings.gradle` in the root as well as the TOML file in `/gradle` and setting `/` as the directory covers both.